opennms-model/src/main/java/org/opennms/netmgt/model/OnmsNode.java

```diff
@@ -323,6 +323,15 @@ public OnmsNode getParent() {
     public void setParent(OnmsNode parent) {
         m_parent = parent;
     }
+    @Transient
+    @XmlAttribute(name = "parent.id")
```

Comment:
Note that `nodeparentid` is already a column in the database, `node` table. I believe we just need to return that. We don't need to access `m_parent` since we are not getting any other data from the parent node.

Reply:
this return m_parent whole object .

    @XmlAttribute(name = "nodeParentID")
    @JsonIgnore
    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="nodeParentID")
    public OnmsNode getParent() {
        return m_parent;
    }
if we want to return on id then the following code .

    @Transient
    @XmlAttribute(name = "parent.id")
    public Integer getParentID() {

        if( m_parent != null ) {
          return m_parent.getId();
         }
          return null;
    }
    
    I think it will be better to follow first code to return whole object.

Reply:
this is returning m_parent whole node .
    @JsonIgnore
    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="nodeParentID")
    @XmlAttribute(name = "nodeParentID")
    public OnmsNode getParent() {
        return m_parent;
    }
If we want id then the following code .

     @XmlAttribute(name = "parent.id")
    public Integer getParentID() {

        if( m_parent != null ) {
          return m_parent.getId();
         }
          return null;
    }

Reply:
I think this will work.  

  // This already maps to the existing column ,readonly
    @XmlAttribute(name = "nodeParentID")
    @Column(name = "nodeParentID",insertable = false, updatable = false)
    public Integer getNodeParentId() {
        return nodeParentId;
    }

    public void setNodeParentId(Integer nodeParentId) {
        this.nodeParentId = nodeParentId;
    }